main/provider/index.ts

```diff
@@ -562,9 +562,13 @@ export class Provider extends EventEmitter {
         typedData = JSON.parse(typedData)
         payload.params = [from, typedData, ...additionalParams]
       } catch (e) {
-        return resError('Malformed typedData.', payload, res)
+        return resError('Malformed typed data', payload, res)
       }
     }
+    
+    if (!Array.isArray(typedData) && !typedData.message) {
```

Comment:
what happens if typed data is an array, will we still display one without a message?

Reply:
The array check is for legacy (non-EIP-712) data which doesn't have a message - this is how that particular case is displayed: https://github.com/floating/frame/blob/4c287dd5ba1ff5f3665b2202369eb5b643bf2e37/app/App/Account/Account/Requests/SignTypedDataRequest/index.js#L96